src/agent/include/agent_registration.hpp

```diff
@@ -17,7 +17,11 @@ namespace agent_registration
     class AgentRegistration
     {
     public:
-        AgentRegistration(std::string user, std::string password, const std::string& key, const std::string& name);
+        AgentRegistration(std::string user,
+                          std::string password,
+                          const std::string& key,
+                          const std::string& name,
+                          const std::string& configPath);
```

Comment:
It would be better if this is an `std::optional` otherwise we force some other uses to pass an empty string.